src/main/java/taskitems/TaskManager.java

```diff
@@ -0,0 +1,236 @@
+        if (type.equals("E")) {
+            tasks.add(new Event(taskName, date));
+        } else {
+            tasks.add(new Deadline(taskName, date));
+        }
+        taskCount++;
+    }
+
+    private void markTask(int number, boolean isBackend) throws IllegalInputException {
+        if (number > taskCount || number < 1) {
+            throw new IllegalInputException();
+        } else {
+            tasks.get(number - 1).setMarked(true);
+        }
+    }
+
+    public void loadData() throws FileNotFoundException {
+        greet.printDecoration();
+        System.out.println("***Please wait while I fetch past data...***");
+        File data = new File(PATH);
+        Scanner reader = new Scanner(data);
+        int count = 1;
+        while (reader.hasNext()) {
+            String line = reader.nextLine();
+            String[] dataLine = line.split(",");
+            boolean hasFailed = false;
+            switch (dataLine[0]) {
+            case "T":
+                try {
+                    addToTasks(dataLine[1],true);
+                } catch (IllegalInputException e) {
+                    System.out.println("Line " + count + " is corrupted.");
+                    hasFailed = true;
+                }
+                break;
+            case "D":
+                try {
+                    addToTasks("D",dataLine[1],dataLine[3],true);
+                } catch (ArrayIndexOutOfBoundsException outOfBoundsException) {
+                    System.out.println("Line " + count + " is corrupted.");
+                    hasFailed = true;
+                }
+                break;
+            case "E":
+                try {
+                    addToTasks("E",dataLine[1],dataLine[3],true);
+                } catch (ArrayIndexOutOfBoundsException outOfBoundsException) {
+                    System.out.println("Line " + count + " is corrupted.");
+                    hasFailed = true;
+                }
+                break;
+            default:
+                System.out.println("Line " + count + " is corrupted,");
+                hasFailed = true;
+                break;
+            }
+            if (!hasFailed && dataLine[2].equals("1")) {
+                try {
+                    markTask(taskCount,true);
+                } catch (IllegalInputException e) {
+                    System.out.println("Corrupted data has been loaded, please check task number: " + taskCount);
+                }
+            }
+            count++;
+        }
+        if (taskCount == 0) {
+            System.out.println("No saved data found");
+        } else {
+            System.out.println("***" + taskCount + " out of " + (count-1) +
+                    " lines of data are valid and fetched accurately." + "***");
+        }
+    }
+
+    public void deleteTask(int number) throws IllegalInputException {
+        greet.printDecoration();
+        if(number > taskCount || number < 1){
```

Comment:
This boolean expression could be [decomposed into intermediate steps](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions) to improve code readability.

Again, the number `1` used here is magic number which can be avoided.